### PR TITLE
[FR] Limit historical rules to the latest 2

### PR DESCRIPTION
**Pull request by Mikaayenson**
## Issues

N/A 

## Summary

Currently we publish historical rules for each release. At times rules may include multiple updates / versions in the release package, which inflates the number of files in the release package. Today the package has > 5000 artifacts, which impacts memory resources upstream. We have been asked by D&R to update this process to release only the latest 2 versions instead of all to de-risk memory concerns.  

I've added a new method `keep_latest_versions` which filters down to the latest two versions. 

## Testing

Debug the `python -m detection_rules dev build-release`

<details><summary>Debugging</summary>
<p>

Screenshot 2024-06-27 at 1 30 50 PM

Screenshot 2024-06-27 at 1 34 21 PM

</p>
</details> 

## Additional Information
- Internal [thread](https://elastic.slack.com/archives/C079Q7CMA2J/p1719504434790359?thread_ts=1719429334.071939&cid=C079Q7CMA2J)

### CODE REVIEW

File: detection_rules/devtools.py [hunk 1]
```diff
@@ -112,8 +112,8 @@ def build_release(config_file, update_version_lock: bool, generate_navigator: bo
                                                            registry_data['conditions']['kibana.version'].strip("^"))
     sde = SecurityDetectionEngine()
     historical_rules = sde.load_integration_assets(previous_pkg_version)
-    historical_rules = sde.transform_legacy_assets(historical_rules)
-    package.add_historical_rules(historical_rules, registry_data['version'])
+    limited_historical_rules = sde.keep_latest_versions(historical_rules)
+    historical_rules = sde.transform_legacy_assets(limited_historical_rules)
```
Comment (Mikaayenson) [inline detection_rules/devtools.py:116]: @terrancedejesus Do we still need to transform or can I remove this?

File: detection_rules/integrations.py [hunk 2]
```diff
@@ -26,6 +26,7 @@
 from .schemas import definitions
 
 MANIFEST_FILE_PATH = get_etc_path('integration-manifests.json.gz')
+NUM_LATEST_RULE_VERSIONS = 2
```
Comment (banderror) [inline detection_rules/integrations.py:29]: Thank you for putting together this PR so quickly @Mikaayenson 🙏 

If we compare the currently published latest version of the package VS a package that would be produced after this optimization:

- what would be the size of each in megabytes?
- what would be the number of assets in each?

**Comment by eric-forte-elastic** [comment]
Testing looks good to me, I have the same debugging results as @Mikaayenson 

<details><summary>Debugging</summary>
<p>

Historical Rules before Limiting
Screenshot from 2024-06-27 14-59-50


Historical Rules post limiting

Screenshot from 2024-06-27 15-00-15

</p>
</details>

File: detection_rules/integrations.py (hunk 2 above)
Comment (banderror) [reply detection_rules/integrations.py:29]: Oh, I just saw in the collapsed "Debugging" section that it's 5553 assets VS 2199 assets after the optimization.

I guess this makes the size in bytes lower by ~50% as well.

**Review by eric-forte-elastic — APPROVED** [review]

File: detection_rules/devtools.py (hunk 1 above)
Comment (terrancedejesus) [reply detection_rules/devtools.py:116]: Yes, we should be able to remove that.

**Review by terrancedejesus — APPROVED** [review]

**Comment by Mikaayenson** [comment]
I just need to double confirm the package is built with the two latest versions and we should be g2g.

**Review by banderror — APPROVED** [review]
LGTM

**Review by shashank-elastic — APPROVED** [review]

**Pull request merged**